gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapper.java

```diff
@@ -0,0 +1,195 @@
+package uk.nhs.adaptors.pss.translator.mapper.medication;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.hl7.fhir.dstu3.model.DateTimeType;
+import org.hl7.fhir.dstu3.model.DomainResource;
+import org.hl7.fhir.dstu3.model.Encounter;
+import org.hl7.fhir.dstu3.model.IdType;
+import org.hl7.fhir.dstu3.model.Medication;
+import org.hl7.fhir.dstu3.model.MedicationRequest;
+import org.hl7.fhir.dstu3.model.MedicationStatement;
+import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.hl7.v3.RCMRMT030101UK04Authorise;
+import org.hl7.v3.RCMRMT030101UK04Component;
+import org.hl7.v3.RCMRMT030101UK04Component2;
+import org.hl7.v3.RCMRMT030101UK04Component3;
+import org.hl7.v3.RCMRMT030101UK04Component4;
+import org.hl7.v3.RCMRMT030101UK04EhrComposition;
+import org.hl7.v3.RCMRMT030101UK04EhrExtract;
+import org.hl7.v3.RCMRMT030101UK04EhrFolder;
+import org.hl7.v3.RCMRMT030101UK04MedicationStatement;
+import org.hl7.v3.RCMRMT030101UK04Prescribe;
+import org.springframework.stereotype.Service;
+
+import lombok.AllArgsConstructor;
+
+import uk.nhs.adaptors.pss.translator.util.DateFormatUtil;
+
+@Service
+@AllArgsConstructor
+public class MedicationRequestMapper {
+    private MedicationMapper medicationMapper;
+    private MedicationRequestOrderMapper medicationRequestOrderMapper;
+    private MedicationRequestPlanMapper medicationRequestPlanMapper;
+    private MedicationStatementMapper medicationStatementMapper;
+
+    public List<DomainResource> mapResources(RCMRMT030101UK04EhrExtract ehrExtract, List<Encounter> encounters, Patient patient) {
+
+        List<RCMRMT030101UK04EhrComposition> ehrCompositions = ehrExtract.getComponent()
+            .stream()
+            .map(RCMRMT030101UK04Component::getEhrFolder)
+            .map(RCMRMT030101UK04EhrFolder::getComponent)
+            .flatMap(List::stream)
+            .map(RCMRMT030101UK04Component3::getEhrComposition)
+            .collect(Collectors.toList());
+
+        for (RCMRMT030101UK04EhrComposition ehrComposition : ehrCompositions) {
+            var context = encounters.stream()
+                .filter(encounter1 -> encounter1.getId().equals(ehrComposition.getId().getRoot())).findFirst();
```

Comment:
Is there a method/util that finds a matching encounter from the ehrCOmpositions? I think there should be by now, maybe use it if there is one or create it for other mappers to use?

Reply:
Only util method I could find returns a reference, this approach requires the full encounter resource.